components/appmgt/org.wso2.carbon.appmgt.hostobjects/src/main/java/org/wso2/carbon/appmgt/hostobjects/DiscoveryClientHostObject.java

```diff
@@ -0,0 +1,246 @@
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.appmgt.hostobjects;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.mozilla.javascript.*;
+import org.wso2.carbon.appmgt.api.APIProvider;
+import org.wso2.carbon.appmgt.api.AppManagementException;
+import org.wso2.carbon.appmgt.impl.APIManagerFactory;
+import org.wso2.carbon.appmgt.impl.clients.AppServerWebappAdminClient;
+import org.wso2.carbon.webapp.mgt.stub.types.carbon.VersionedWebappMetadata;
+import org.wso2.carbon.webapp.mgt.stub.types.carbon.WebappMetadata;
+import org.wso2.carbon.webapp.mgt.stub.types.carbon.WebappsWrapper;
+
+/**
+ * Host object to expose discovery service
+ * TODO: Move this to factory service and do away with "HostObject" approach.
+ */
+public class DiscoveryClientHostObject extends ScriptableObject {
+
+    private static final Log log = LogFactory.getLog(DiscoveryClientHostObject.class);
+
+    private static final int MAX_GENERATED_CONTEXT_SUFFIX = 1000;
+    private static final String CONTEXT_NOT_GENERATED = "<could-not-generate>";
+    private static final String APP_PROPERTY_NAME = "name";
+    private static final String APP_PROPERTY_VERSION = "version";
+    private static final String APP_PROPERTY_APP_TYPE = "appType";
+    private static final String APP_PROPERTY_CONTEXT = "context";
+    private static final String APP_PROPERTY_PROXY_CONTEXT = "proxyContext";
+    private static final String APP_PROPERTY_STATUS = "status";
+    private static final String APP_PROPERTY_STATUS_NEW = "New";
+    private static final String APP_PROPERTY_WEB_APP_ID = "webappId";
+    private static final String APP_PROPERTY_ID = "id";
+    private static final String APP_PROPERTY_HOST = "host";
+    private static final String APP_PROPERTY_PROTOCOL = "protocol";
+    private static final String APP_PROPERTY_PORT = "port";
+    private static final String APP_PROPERTY_HTTP_URL = "httpUrl";
+    //    private static final String APP_PROPERTY_HOST = "host";
+    private static final String APP_PROPERTY_STATUS_PROXY_CREATED = "ProxyCreated";
+
+    @Override
+    public String getClassName() {
+        return "DiscoveryClient";
+    }
+
+    // The zero-argument constructor used for create instances for runtime
+    public DiscoveryClientHostObject() throws AppManagementException {
+    }
+
+    /**
+     * Calls he backend AS and discover available applications.
+     *
+     * @param cx
```

Comment:
Agreed. However, the usage of this class was a temporary work(as per the PR comment), such that Jaggery side UI to be ready for demo. This has been superseded by REST service "DiscoveryService" subsequently(before product v1.0.0 get released)